src/main/java/no/digipost/security/cert/ProdEnvCertificates.java

```diff
@@ -0,0 +1,175 @@
+     */
+    public static X509Certificate buypassClass3RootCaG2SoftToken() {
+        return BuypassClass3RootCaG2ST.cert;
+    }
+
+    /**
+     * Buypass SEID v2.0 Enterprise certificate issuer CA G2 'Soft Token' Business
+     *
+     * <p>CN=Buypass Class 3 CA G2 ST Business, O=Buypass AS, OID.2.5.4.97=NTRNO-983163327, C=NO
+     * <p>valid 2020-11-10 to 2040-11-10
+     */
+    public static X509Certificate buypassClass3CaG2SoftToken() {
+        return BuypassClass3CaG2ST.cert;
+    }
+
+    /**
+     * Buypass Class 3 Root CA G2 'Hard Token'
+     *
+     * <p>CN=Buypass Class 3 Root CA G2 HT, O=Buypass AS, OID.2.5.4.97=NTRNO-983163327, C=NO
+     * <p>valid 2020-11-10 to 2045-11-10
+     */
+    public static X509Certificate buypassClass3RootCaG2HardToken() {
+        return BuypassClass3RootCaG2HT.cert;
+    }
+
+    /**
+     * Buypass SEID v2.0 Enterprise certificate issuer CA G2 'Hard Token'
+     *
+     * <p>CN=Buypass Class 3 CA G2 HT Business, O=Buypass AS, OID.2.5.4.97=NTRNO-983163327, C=NO
+     * <p>valid 2020-11-10 to 2040-11-10
+     */
+    public static X509Certificate buypassClass3CaG2HardToken() {
+        return BuypassClass3CaG2HT.cert;
+    }
+
+
+    /**
+     * Buypass Root CA (SEID generation 1)
+     *
+     * <p>CN=Buypass Class 3 Root CA, O=Buypass AS-983163327, C=NO
+     * <p>valid 2010-10-26 to 2040-10-26
+     */
+    public static X509Certificate buypassClass3RootCa() {
+        return BuypassClass3RootCa.cert;
+    }
+
+    /**
+     * Buypass certificate issuer CA (SEID generation 1)
+     *
+     * <p>CN=Buypass Class 3 CA 3, O=Buypass AS-983163327, C=NO
+     * <p>valid 2012-09-25 to 2032-09-25
+     */
+    public static X509Certificate buypassClass3Ca3() {
+        return BuypassClass3Ca3.cert;
+    }
+
+    /**
+     * Commfides Root CA (SEID generaation 1)
+     *
+     * <p>C=NO, O=Commfides Norge AS - 988 312 495, OU=Commfides Trust Environment (c) 2011 Commfides Norge AS, CN=CPN RootCA SHA256 Class 3
+     * <p>valid 2011-02-28 to 2025-01-01
+     */
+    public static X509Certificate commfidesRootCa() {
+        return CommfidesRootCa.cert;
+    }
+
+    /**
+     * Commfides certificate issuer CA (SEID generation 1)
+     *
+     * <p>C=NO, O=Commfides Norge AS - 988 312 495, OU=Commfides Trust Environment (c) 2011 Commfides Norge AS, CN=CPN Enterprise SHA256 CLASS 3
+     * <p>valid 2011-02-28 to 2024-12-31
+     */
+    public static X509Certificate commfidesCa() {
+        return CommfidesCa.cert;
+    }
+
+    /**
+     * Digipost Root CA
+     *
+     * <p>CN=Digipost Root CA, OU=Digipost, O=Posten Norge AS, ST=Oslo, C=NO
+     * <p>valid 2017-10-06 to 2050-01-01
+     */
+    public static X509Certificate digipostRootCa() {
+        return DigipostRootCa.cert;
+    }
+
+
+    // X509Certificate singletons
+
+    private static final class BuypassClass3RootCa {
+        static final X509Certificate cert = readCertificate("BPClass3RootCA.cer");
+    }
```

Comment:
Just some explanation of this pattern.
Why all the private nested classes, each with one field?

This is to ensure that each certificate are not read into memory unless explicitly requested, and to ensure it is read at most once (singletons). If they were fields on `ProdEnvCertificates`, all fields would be initialized on the first reference (and thus loading of) the `ProdEnvCertificates` class. Having public static methods above each refer to the class containing the field for the certificate will defer loading of each of these "singleton classes" until the respective public methods above are invoked. And the classloading mechanism of the JVM guaranties thread-safe initializing of these fields, instead of having to code this oneself, with some traditional "singleton pattern" that may or may not expect certificates to be acquired by multiple threads at once.

https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom